pilot/pkg/features/pilot.go

```diff
@@ -173,7 +173,9 @@ var (
 	EnableProtocolSniffing = env.RegisterBoolVar(
 		"PILOT_ENABLE_PROTOCOL_SNIFFING",
 		false,
-		"If enabled, protocol sniffing will be used on ports whose port protocol is not specified or unsupported",
+		"If enabled, protocol sniffing will be used on ports whose port protocol is not specified or unknown."+
+			"The listener will detect http vs tcp automatically. Protocol sniffing won't work on ports which are "+
+			"used by MySQL and MongoDB. Protocol sniffing are not supported for ports using HTTP2 and gRPC.",
```

Comment:
It won't work on them? Or it will not detect mysql/mongo?

> Protocol sniffing are not supported for ports using HTTP2 and gRPC.

What is not supported mean? Treated as TCP? Breaks everything?

I thought sniffing supported http2

Reply:
> What is not supported mean? Treated as TCP? Breaks everything?

Envoy http inspector supports inspecting HTTP2 but pilot implementation doesn't. If h2 traffic comes in, it will hit the http filter chain. The HTTP 1.x codec will try to decode http2 traffic, which will fail.

The reason that h2 is not supported is HTTP2 listener and cluster is different from HTTP1. It would be more complicated to support HTTP2. I would prefer to have a separate to enable HTTP2 and gRPC.

Reply:
Protocol sniffing will be disabled on MySQL and Mongo ports. HTTP inspector also cannot detect MySQL/Mongo.

Reply:
I suggest to not add info here. No one reads these things, even though they get published in the docs. Add this to the FAQs section in istio.io. Also add this to the "adding sidecar" section, where we are asking people to name the ports properly.